matchbox/fixture_test.go

```diff
@@ -0,0 +1,245 @@
+package matchbox
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"net"
+
+	"google.golang.org/grpc"
+
+	"github.com/coreos/matchbox/matchbox/rpc"
+	"github.com/coreos/matchbox/matchbox/server"
+	"github.com/coreos/matchbox/matchbox/storage"
+)
+
+type FixtureServer struct {
+	Store    storage.Store
+	Server   *grpc.Server
+	Listener net.Listener
+}
+
+func NewFixtureServer(s storage.Store) *FixtureServer {
+	return &FixtureServer{
+		Store: s,
+	}
+}
+
+func (s *FixtureServer) Start() {
+	var err error
+	s.Listener, err = net.Listen("tcp", "127.0.0.1:0")
```

Comment:
just like this:

```
panic: failed to start listener: listen tcp 127.0.0.1:8080: bind: address already in use [recovered]
        panic: failed to start listener: listen tcp 127.0.0.1:8080: bind: address already in use

goroutine 417 [running]:
```

This is  produced because the stop and start was that fast that the system was unable to release the port.